js/bitmex.js

```diff
@@ -2666,6 +2697,46 @@ module.exports = class bitmex extends Exchange {
         return await this.privatePostPositionIsolate (this.extend (request, params));
     }
 
+    async fetchDepositAddress (code, params = {}) {
+        /**
+         * @method
+         * @name bitmex#fetchDepositAddress
+         * @description fetch the deposit address for a currency associated with this account
+         * @see https://www.bitmex.com/api/explorer/#!/User/User_getDepositAddress
+         * @param {string} code unified currency code
+         * @param {object} params extra parameters specific to the bitmex api endpoint
+         * @param {string} params.network deposit chain, can view all chains via this.publicGetWalletAssets, default is eth, unless the currency has a default chain within this.options['networks']
+         * @returns {object} an [address structure]{@link https://docs.ccxt.com/en/latest/manual.html#address-structure}
+         */
+        await this.loadMarkets ();
+        const networkCode = this.safeStringUpper (params, 'network');
+        if (networkCode === undefined) {
+            throw new ArgumentsRequired (this.id + ' fetchDepositAddress requires params["network"]');
+        }
+        const networks = this.safeValue (this.options, 'networks', {});
+        const networkId = this.safeStringLower (networks, networkCode, networkCode);
+        const currency = this.currency (code);
+        let currencyId = currency['id'];
+        const idLength = currencyId.length;
+        currencyId = currencyId.slice (0, idLength - 1) + currencyId.slice (idLength - 1, idLength).toLowerCase ();  // make the last letter lowercase
+        params = this.omit (params, 'network');
+        const request = {
+            'currency': currencyId,
+            'network': networkId,
+        };
+        const response = await this.privateGetUserDepositAddress (this.extend (request, params));
```

Comment:
@samgermain something like `fetchDepositAddress(USDT, {"network":"ERC20"})` fails. Can you take a look pls?

```
bitmex.fetchDepositAddress (USDT, [object Object])
BadRequest bitmex {"error":{"message":"Invalid network for currency","name":"HTTPError"}}
---------------------------------------------------
[BadRequest] bitmex {"error":{"message":"Invalid network for currency","name":"HTTPError"}}

    at handleErrors               js/bitmex.js:2767                     throw new BadRequest (feedback);                                                
    at                            js/base/Exchange.js:599               const skipFurtherErrorHandling = this.handleErrors (response.status, response.s…
    at processTicksAndRejections  node:internal/process/task_queues:95                                                                                  
    at async timeout              js/base/functions/time.js:181         return await Promise.race ([promise, expires.then (() => { throw new TimedOut (…
    at fetch2                     js/base/Exchange.js:1967              return await this.fetch (request['url'], request['method'], request['headers'],…
    at request                    js/base/Exchange.js:1971              return await this.fetch2 (path, api, method, params, headers, body, config, con…
    at fetchDepositAddress        js/bitmex.js:2727                     const response = await this.privateGetUserDepositAddress (this.extend (request,…
    at async run                  examples/js/cli.js:281                const result = await exchange[methodName] (... args)                            

bitmex {"error":{"message":"Invalid network for currency","name":"HTTPError"}}
```

Reply:
@samgermain can you check this comment pls

Reply:
Should be good now

Reply:
@samgermain will check it thanks